pages/index.js

```diff
@@ -12,6 +12,8 @@ export default function Home({ locationsInfo}) {
 const { data: session } = useSession() 
 const { data, isLoading, error } = useSWR("/api/locations");
 
+const visibleData = data.filter((visibleLocation) => visibleLocation.visible === true)
```

Comment:
This works perfectly fine. The next step would be to filter locations in the backend and only pass along the filtered locations. Currently, the entire data is sent to the client (and I could inspect it in Dev Tools even if you don't show it). Filtering in the backend would eliminate that possibility and save bandwith too.